modules-packages-crates/includes/1-scoping-terminology.md

```diff
@@ -0,0 +1,101 @@
+# Scoping Terminology
+
+Before we start, it is important to explain the concepts behind code organization in Rust programs:
+
+- A **package** is one or more **crates** that provide a set of functionality. A **package** contains a `Cargo.toml` file that describes how to build those **crates**.
+- A **crate** is a compilation unit, which is the smallest amount of code that the Rust compiler can operate on. Once compiled, it will produce either an executable or a library.
+- A **crate** contains an implicit, un-named top-level **module**.
+- A **module** is a *(possibly nested)* unit of code organization inside a **crate**.
+- Recursive definitions can span **modules**, but not **crates**.
+
+## Package
+
+Whenever we run the command `$ cargo new <project-name>`, Cargo creates a package for us:
+
+    $ cargo new my-project
+         Created binary (application) `my-project` package
+
+Here, we have a package that only contains `src/main.rs`, meaning it only contains a binary crate named `my-project`:
+
+    my-project
+    ├── src
+    │  └── main.rs
+    └── Cargo.toml
+
+A package can have multiple binary crates by placing files in the `src/bin` directory: each file will be a separate binary crate.
+
+If a package contains `src/main.rs` and `src/lib.rs`, it has two crates: a library and a binary, both with the same name as the package.
+
+## Crates
+
+Rust's compilation model centers on artifacts called crates, that can be compiled into a binary or into a library.
+
+## Library Crates
+
+We've already covered how to create a binary program, and creating a library is just as easy. To create a library, pass the `--lib` command line parateter to the `cargo new` command:
+
+    $ cargo new --lib my-library
+         Created library `my-library` package
+
+You can see that, instead of a `src/main.rs` file, you now get a =src/lib.rs' file.
+
+    my-library
+    ├── src
+    │  └── lib.rs
+    └── Cargo.toml
+
+When you tell Cargo to complie this crate, you'll get a library file called `libmy_library.rlib` that can be published and linked to other projects.
+ 
+## Modules
+
+Rust provides a powerful module system that can be used to hierarchically split code into logical units and that ease readability and reuse.
+
+A module is a collection of items:
+
+- Constants
+- Type aliases
+- Functions
+- Structs
+- Enums
+- Traits
+- `impl` blocks
+- Other modules, which can nest arbitrarily.
```

Comment:
This last point is a bit confusing. I'm not sure I understand about the arbitrary nesting. Can you rephrase?

Maybe it can just say "other modules"